packages/components/table/__tests__/table-column.test.ts

```diff
@@ -633,6 +633,150 @@ describe('table column', () => {
+            this.expandCount++
+          },
+          refreshData() {
+            this.testData = getTestData()
+          },
+        },
+      })
+    }
+
+    it('should render expand column correctly', async () => {
+      const wrapper = createInstance()
+      await doubleWait()
+      expect(wrapper.findAll('td.el-table__expand-column').length).toEqual(5)
+      wrapper.unmount()
+    })
+
+    describe('preserve-expanded-content', () => {
+      it('should lose expanded state when refreshing data without preserve-expanded-content', async () => {
+        const wrapper = createInstance() // No preserve-expanded-content prop
+        await doubleWait()
+
+        // Click expand icon and verify initial state
+        const expandIcon = wrapper.find('.el-table__expand-icon')
+        await expandIcon.trigger('click')
+        await doubleWait()
+        expect(wrapper.findAll('.el-table__expanded-cell')).toHaveLength(1)
+
+        // Clear and replace data
+        wrapper.vm.testData = []
+        await doubleWait()
+        wrapper.vm.testData = getTestData()
+        await doubleWait()
+
+        // Without preserve-expanded-content, expanded state should be lost
+        expect(wrapper.findAll('.el-table__expanded-cell')).toHaveLength(0)
+        // All expand icons should be in collapsed state
+        expect(
+          wrapper.find('.el-table__expand-icon--expanded').exists()
+        ).toBeFalsy()
+
+        wrapper.unmount()
+      })
+
+      it('should preserve expanded state when refreshing data with preserve-expanded-content', async () => {
+        const wrapper = createInstance('preserve-expanded-content')
+        await doubleWait()
+
+        // First expand the row
+        const expandEl = wrapper.find('.el-table__expand-icon')
+        await expandEl.trigger('click')
+
+        // Verify initial expand
+        expect(wrapper.find('.el-table__expanded-cell').exists()).toBeTruthy()
+        const expandedContent = wrapper.find('.el-table__expanded-cell').text()
+        expect(expandedContent).toContain(wrapper.vm.testData[0].name)
+
+        // Refresh data
+        wrapper.vm.refreshData()
+        await doubleWait()
+
+        // With preserve-expanded-content, expanded state and content should persist
+        expect(wrapper.find('.el-table__expanded-cell').exists()).toBeTruthy()
+        expect(wrapper.find('.el-table__expanded-cell').text()).toBe(
+          expandedContent
+        )
+
+        wrapper.unmount()
+      })
+
+      it('should preserve multiple expanded rows when refreshing data', async () => {
+        const wrapper = createInstance('preserve-expanded-content')
+        await doubleWait()
+
+        // Find first two expand icons
+        const expandIcons = wrapper
+          .findAll('.el-table__expand-icon')
+          .slice(0, 2)
+
+        // Expand first two rows
+        for (const icon of expandIcons) {
+          await icon.trigger('click')
+          await doubleWait()
+        }
```

Comment:
The code here may be adjusted
```suggestion
        for (const icon of expandIcons) {
          icon.trigger('click')
        }
        await doubleWait()
```